step/step.go

```diff
@@ -1,42 +1,72 @@
 package step
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
+
+	"github.com/stackpulse/steps-sdk-go/env"
 )
 
 const (
-	ExitCodeOK      = 0
-	ExitCodeFailure = 1
+	stepErrorKey = "step_error"
 )
 
 type Outputs struct {
 	Object interface{} `json:"api_object,omitempty"`
 }
 
+type JSONOutput map[string]interface{}
+
 // An interface describing a general purpose stackpulse step
 type Step interface {
-	Init() error
-	Run() (exitCode int, output []byte, err error)
+	Init() *Error
+	Run() (JSONOutput, *Error)
 }
 
-// A helper method that execute a given step according to step lifecycle
-func Run(step Step) {
-	err := step.Init()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed init step arguments, %v\n", err)
-		os.Exit(ExitCodeFailure)
+func printOutput(stepOutput map[string]interface{}, stepError *Error) {
```

Comment:
```suggestion
func printOutput(stepOutput map[string]interface{}, stepError *Error) error {
```

the function should return an error so run can return `ExitCodeFailure` if the marshalling fails